src/com/google/javascript/jscomp/js/es6/util/inherits.js

```diff
@@ -49,25 +49,31 @@
  *
  * @param {!Function} childCtor Child class.
  * @param {!Function} parentCtor Parent class.
+ * @param {boolean} es5Out language out is ES5
  */
-$jscomp.inherits = function(childCtor, parentCtor) {
-  /** @constructor */
-  function tempCtor() {}
-  tempCtor.prototype = parentCtor.prototype;
-  childCtor.superClass_ = parentCtor.prototype;
-  childCtor.prototype = new tempCtor();
-  /** @override */
-    childCtor.prototype.constructor = childCtor;
+$jscomp.inherits = function(childCtor, parentCtor, es5Out) {
+  if (es5Out === true && Object.setPrototypeOf) {
```

Comment:
I discussed this a bit with @MatrixFrog today.
I think what's needed is a `--use_setprototypeof_for_inheritance` option to guard this.
If that option isn't given, we pass `false` to `$jscomp.inherits` as its `useSetPrototypeOf` parameter.
`$jscomp.inherits` throws an error if the argument is true, but `Object.setPrototypeOf` doesn't exist.